libraries/AC_AttitudeControl/AC_PosControl.cpp

```diff
@@ -775,23 +774,23 @@ void AC_PosControl::input_vel_accel_z(Vector3f& vel, const Vector3f& accel, bool
     }
 
     // adjust desired alt if motors have not hit their limits
-    update_pos_vel_accel_z(_pos_target, _vel_desired, _accel_desired, _dt, _limit_vector);
+    update_pos_vel_accel(_pos_target.z, _vel_desired.z, _accel_desired.z, _dt, _limit_vector.z);
 
-    shape_vel_accel_z(vel, accel,
-        _vel_desired, _accel_desired,
-        _vel_max_down_cms, _vel_max_up_cms,
-        -constrain_float(accel_z_cmss, 0.0f, 750.0f), accel_z_cmss,
-        _tc_z_s, _dt);
+    shape_vel_accel(vel, accel,
+                    _vel_desired.z, _accel_desired.z,
+                    _vel_max_down_cms, _vel_max_up_cms,
+                    -constrain_float(accel_z_cmss, 0.0f, 750.0f), accel_z_cmss,
+                    _tc_z_s, _dt);
 
-    update_vel_accel_z(vel, accel, _dt, Vector3f());
+    update_vel_accel(vel, accel, _dt, 0);
 }
 
 /// set_pos_target_z_from_climb_rate_cm - adjusts target up or down using a commanded climb rate in cm/s
 ///     using the default position control kinimatic path.
 void AC_PosControl::set_pos_target_z_from_climb_rate_cm(const float vel, bool force_descend)
 {
-    Vector3f vel_3f = Vector3f{0.0f, 0.0f, vel};
-    input_vel_accel_z(vel_3f, Vector3f{0.0f, 0.0f, 0.0f}, force_descend);
+    float vel2 = vel;
+    input_vel_accel_z(vel2, 0, force_descend);
```

Comment:
@lthall this one needs looking at, should it be altering vel in caller?

Reply:
No, in this use case the velocity command is our final velocity target with zero acceleration. We only need to update the velocity if we are targeting an instantaneous velocity with a non-zero acceleration.